src/analyses/variable-sensitivity/constant_array_abstract_object.cpp

```diff
@@ -0,0 +1,580 @@
+
+ Outputs: An abstract object pointer of type type().subtype() (i.e. the
+          type of the array's values).
+
+ Purpose: Short hand method for creating a top element of the array
+
+\*******************************************************************/
+
+abstract_object_pointert constant_array_abstract_objectt::get_top_entry(
+  const abstract_environmentt &env,
+  const namespacet &ns) const
+{
+  return env.abstract_object_factory(type().subtype(), ns, true, false);
+}
+
+/*******************************************************************\
+
+Function: constant_array_abstract_objectt::eval_index
+
+  Inputs:
+   environment - the abstract environment
+   ns - the namespace
+
+ Outputs: An abstract object pointer of type type().subtype() (i.e. the
+          type of the array's values).
+
+ Purpose: Short hand method for creating a top element of the array
+
+\*******************************************************************/
+
+bool constant_array_abstract_objectt::eval_index(
+  const index_exprt &index,
+  const abstract_environmentt &env,
+  const namespacet &ns,
+  mp_integer &out_index) const
+{
+  abstract_object_pointert index_abstract_object = env.eval(index.index(), ns);
+  exprt value = index_abstract_object->to_constant();
+  if(value.is_constant())
+  {
+    constant_exprt constant_index = to_constant_expr(value);
+    bool result = to_integer(constant_index, out_index);
+    return !result;
+  }
+  else
+  {
+    return false;
+  }
+}
+
+/**
+ * Apply a visitor operation to all sub elements of this abstract_object.
+ * A sub element might be a member of a struct, or an element of an array,
+ * for instance, but this is entirely determined by the particular
+ * derived instance of abstract_objectt.
+ *
+ * \param visitor an instance of a visitor class that will be applied to
+ * all sub elements
+ * \return A new abstract_object if it's contents is modifed, or this if
+ * no modification is needed
+ */
+abstract_object_pointert constant_array_abstract_objectt::visit_sub_elements(
+  const abstract_object_visitort &visitor) const
+{
+  const auto &result =
+    std::dynamic_pointer_cast<constant_array_abstract_objectt>(mutable_clone());
+
+  bool modified = false;
+
+  shared_array_mapt::viewt view;
+  result->map.get_view(view);
+
+  for(auto &item : view)
+  {
+    auto newval = visitor.visit(item.second);
+    if(newval != item.second)
+    {
+      result->map.replace(item.first, std::move(newval));
+      modified = true;
+    }
+  }
+
+  if(modified)
+  {
+    return result;
+  }
+  else
+  {
+    return shared_from_this();
+  }
+}
+
+void constant_array_abstract_objectt::get_statistics(
+  abstract_object_statisticst &statistics,
+  abstract_object_visitedt &visited,
+  const abstract_environmentt &env,
+  const namespacet &ns) const
+{
+  array_abstract_objectt::get_statistics(statistics, visited, env, ns);
+  shared_array_mapt::viewt view;
+  map.get_view(view);
+  for(auto const &object : view)
```

Comment:
⛏️ Preferred for consistency with the rest of the codebase:
```suggestion
  for(const auto &object : view)
```

Reply:
Fixed.